src/cow-react/common/hooks/useIsSmartContractWallet.ts

```diff
@@ -30,13 +31,15 @@ export function useIsSmartContractWallet(): boolean {
 
   const isArgentWallet = useIsArgentWallet()
   const isSmartContract = useCheckIsSmartContract()
+  const isAmbireWallet = useIsAmbireWallet()
 
   useEffect(() => {
     if (!account) {
       setIsSmartContractWallet(false)
-    } else if (account && isArgentWallet) {
-      setIsSmartContractWallet(true)
-    } else if (account && isSmartContract) {
+      return
+    }
+
+    if (isAmbireWallet || isArgentWallet || isSmartContract) {
       setIsSmartContractWallet(true)
```

Comment:
Just curious, why this is needed? 

`isSmartContract` will tell us if the address has code, therefore if its a smart contract.
Why do we need to heck if its ambire wallet?

Reply:
Because this hooks `useCheckIsSmartContract` will actually return false for Ambire and then we have an issue with the Network selector showing up that we don't want at this point for this wallet.